artifactory/services/go/publishzipandmod.go

```diff
@@ -30,35 +37,39 @@ func (pwa *publishZipAndModApi) PublishPackage(params GoParams, client *rthttpcl
 	if err != nil {
 		return err
 	}
-	zipUrl := url
+	pwa.clientDetails = ArtDetails.CreateHttpClientDetails()
+	pwa.client = client
 	moduleId := strings.Split(params.GetModuleId(), ":")
-	err = createUrlPath(moduleId[0], params.GetVersion(), params.GetProps(), ".zip", &zipUrl)
+	// Upload zip file
+	err = pwa.upload(params.GetZipPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".zip", url)
 	if err != nil {
 		return err
 	}
-	clientDetails := ArtDetails.CreateHttpClientDetails()
-
-	addGoVersion(params, &zipUrl)
-	resp, _, err := client.UploadFile(params.GetZipPath(), zipUrl, "", &clientDetails, GoUploadRetries, nil)
+	// Upload mod file
+	err = pwa.upload(params.GetModPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".mod", url)
 	if err != nil {
 		return err
 	}
-	err = errorutils.CheckResponseStatus(resp, http.StatusCreated)
-	if err != nil {
-		return err
+	if version.NewVersion(ArtifactoryMinSupportedVersionForInfoFile).AtLeast(pwa.artifactoryVersion) && params.GetInfoPath() != "" {
+		// Upload info file. This supported from Artifactory version 6.10.0 and above
+		return pwa.upload(params.GetInfoPath(), moduleId[0], params.GetVersion(), params.GetProps(), ".info", url)
 	}
-	err = createUrlPath(moduleId[0], params.GetVersion(), params.GetProps(), ".mod", &url)
+	return nil
+}
+
+func addGoVersion(version string, urlPath *string) {
+	*urlPath += ";go.version=" + url.QueryEscape(version)
+}
+
+func (pwa *publishZipAndModApi) upload(localPath, moduleId, version, props, ext string, urlPath string) error {
```

Comment:
The arguments type *string* appears twice, instead of once in the method signature.
I also suggest addign a description for the arguments, specifically for **ext**.

Reply:
Done